purchase_stock_no_bill_valuation/models/account_move_line.py

```diff
@@ -0,0 +1,9 @@
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    def _apply_price_difference(self):
```

Comment:
Para analizar, si es que hay casos de uso que no sean tan contundentes como para anular al 100% la funcionalidad de cálculo de Odoo en este sentido, sino de incluir algun tipo de parametrización que mediante el método `_get_values_in_moves`. Por ejemplo: a nivel del producto o de la categoría. En ese caso, 
quizá se podría heredar este otro método (quitando en su totalidad `_apply_price_difference`, por ejemplo haciendo algo así:
```suggestion
    def _get_valued_in_moves(self):
        self.ensure_one()
        return self.purchase_line_id.move_ids.filtered(
            lambda m: m.state == 'done' and m.product_qty != 0 and m.categ_id.not_valuation_price_difference)
    
```

Reply:
Ahí subí unos cambios, te parece mejor así?